File: 387.first-unique-character-in-a-string.py
#
# @lc app=leetcode id=387 lang=python3
#
# [387] First Unique Character in a String
#

# @lc code=start
from typing import *
class Solution:
    def firstUniqChar(self, s: str) -> int:
        count_map = {}
        for c in s:
            if not c in count_map:
                count_map[c] = 0
            count_map[c] += 1
        for index,c in enumerate(s):
            if count_map[c] == 1:
                return index 
        return -1

        
# @lc code=end

if __name__ == "__main__":
    print(Solution().firstUniqChar("leetcode"))

File: golang/387.first-unique-character-in-a-string.go
/*
 * @lc app=leetcode.cn id=387 lang=golang
 *
 * [387] First Unique Character in a String
 */
package main

// @lc code=start
func firstUniqChar(s string) int {
	existMap := map[byte]int{}
	for i := len(s) - 1; i >= 0; i-- {
		char := s[i]
		existMap[char] += 1
	}
	for i := 0; i < len(s); i++ {
		if existMap[s[i]] == 1 {
			return i
		}
	}
	return -1
}

// @lc code=end
